mlir/lib/Dialect/Bufferization/Transforms/BufferResultsToOutParams.cpp

```diff
@@ -224,7 +232,13 @@ LogicalResult mlir::bufferization::promoteBufferResultsToOutParams(
       return failure();
     }
   }
-  if (failed(updateCalls(module, options)))
+  auto defaultAllocationFn = [](OpBuilder &builder, Location loc,
```

Comment:
Maybe we could give keep this as a default value in the struct, then the field doesn't have to be optional and you can pass the `options` to `updateCalls`. Not sure if it's possible without including `MemRefOps.h` in  `Passes.h` though. (Maybe we can split the header?)

Reply:
sure that sounds good to me. i think i would want to do that with the copy op too

Reply:
i was just kinda following what had already been done with copy. i think they should be treated the same in this context either way

Reply:
> Not sure if it's possible without including MemRefOps.h in Passes.h though. (Maybe we can split the header?)

If you are opposed to including `MemRefOps.h` I could just forward declare the appropriate ops if you prefer. I have no particular preference and honestly don't know what's best practice in this case.

Reply:
i went ahead and just included `MemRef.h` for now. `Passes.h` was already included. Naively, I think this is ok since the default use of this would have to link the memref dialect anyway if I'm not mistaken. But let me know if this is no good